app/libs/loaders/Leaf_autoloader.php

```diff
@@ -23,12 +23,17 @@
 $site = new App\Leaf\Site($file_paths_db, $_SERVER['SCRIPT_FILENAME']);
 
 if ($site->error) {
-    throw new Exception("Sorry the page you are looking for could not be found, please check the url and try again.");

```

Comment:
Note that http_response_code() does not abort processing, and removing the throw substantially changes program flow. Not sure if this was intended.

Reply:
When I tested the throw on staging it threw a 500 with no real message, that is why I changed it to this and added the die() right after it. This way the nginx will capture the 400 and send them to a 4xx page.